python/pyspark/sql/dataframe.py

```diff
@@ -271,11 +279,38 @@ def explain(self, extended=False):
         ...
         == Physical Plan ==
         ...
+
+        >>> df.explain(mode="formatted")
+        == Physical Plan ==
+        * Scan ExistingRDD (1)
+        <BLANKLINE>
+        <BLANKLINE>
```

Comment:
I think you can just remove these BLANKLINE since `NORMALIZE_WHITESPACE` is enabled (see the bottom of this file).